localstack/config.py

```diff
@@ -590,9 +590,9 @@ def in_docker():
     except Exception:
         pass
     with open("/proc/1/cgroup", "rt") as ifh:
-        os_hostname = open("/etc/hostname", "rt").read().strip()
+        os_hostname = socket.gethostname()
         content = ifh.read()
-        if os_hostname in content or "docker" in content:
+        if (os_hostname and os_hostname in content) or "docker" in content:
```

Comment:
nit: we could apply a tiny optimization - first checking if `docker` is contained in `content`, and only then calling `gethostname()`..
```
content = ifh.read()
if "docker" in content:
    return True
os_hostname = socket.gethostname()
...
```

Reply:
Done :)